### PR TITLE
Issue 2 check ownership groups

### PR DESCRIPTION
**Pull request by dmgolembiowski**
Added a feature extension to check the group and owner bits on the file in question. The feature is available for Unix OS's.

### CODE REVIEW

**Comment by dmgolembiowski** [comment]
For #2

**Comment by dmgolembiowski** [comment]
@fitzgen Would you be able to offer any guidance or suggestions on what I need to do to make this pass? This is my first open source PR, so I'm a little lost.

File: Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,6 @@ winapi = { version = "0.3", features = ["winbase"] }
 
 [dev-dependencies]
 diff = "0.1.10"
+
+[dependencies]
```
Comment (alpha-tango-kilo) [inline Cargo.toml:22]: This dependency should be Unix only, see [this section](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies) of the Cargo book for a how to
```suggestion
[target.'cfg(unix)'.dependencies]
```

File: src/lib.rs [hunk 2]
```diff
@@ -79,12 +84,45 @@ pub trait IsExecutable {
     fn is_executable(&self) -> bool;
 }
 
+/// Returns `Result<Path, io::Error>` if there is a file at the given path and the
+/// current run-level is permitted to execute it.
+///
+/// See the module documentation for details.
+#[cfg(unix)]
+pub fn is_permitted<P>(path: P) -> Result<::std::path::PathBuf, io::Error>
```
Comment (alpha-tango-kilo) [inline src/lib.rs:92]: Use the full path for `io` here and get rid of the import on line 58, it's unneeded unless compiling for Unix

File: src/lib.rs [hunk 3]
```diff
@@ -68,6 +71,8 @@ where
     path.as_ref().is_executable()
 }
 
+
```
Comment (alpha-tango-kilo) [inline src/lib.rs:74]: Adding random whitespace isn't good practice for creating a PR. You don't want to change anything more than necessary

File: src/lib.rs [hunk 4]
```diff
@@ -96,6 +134,46 @@ mod unix {
...
+    ///  Assuming that the set of all groups shared by the user, 
+    ///  (G* := { Gn, Gn+1, Gn+2, ...}), is a superset of the set whose only
+    ///  entry is the user's GID, (G_user := G* - Gn).
+    ///  Then, Ǝ a possibility some arbitary GID, like (Gn+14) for example,
+    ///  happens to be the GID belonging to group on that file.
+    ///  
+    ///  In other words, we'll collect each of the (G*) entries
+    ///  and see if there's a match. Otherwise, we check who owns the file and
+    ///  perform a similar check.
+    impl IsPermitted for Path {
```
Comment (alpha-tango-kilo) [inline src/lib.rs:151]: The formatting within this `impl` is questionable. I'd recommend trying to use `rustfmt` if you can do without massively changing other parts of the lib, or tidying this up yourself by hand

File: tests/tests.rs [hunk 5]
```diff
@@ -1,7 +1,7 @@
 extern crate diff;
 extern crate is_executable;
 
-use is_executable::is_executable;
+use is_executable::{ is_executable, is_permitted };
```
Comment (alpha-tango-kilo) [inline tests/tests.rs:4]: This import won't work when compiling for not-Unix. Importing `is_permitted` must be done in `mod unix`

File: src/lib.rs [hunk 6]
```diff
@@ -79,12 +84,45 @@ pub trait IsExecutable {
...
+    P: AsRef<Path>
+{
+    path.as_ref().is_permitted()
+}
+
+/// An extension trait for `std::fs::Path` providing an `is_permitted` method.
+///
+/// See the module documentation for examples.
+#[cfg(unix)]
+pub trait IsPermitted {
```
Comment (alpha-tango-kilo) [inline src/lib.rs:103]: Both this trait declaration and `is_permitted` might as well go into the `mod unix`, You may need to add a `pub use` statement to re-export what's needed to the base of the crate if that's what you're after

**Review by alpha-tango-kilo — CHANGES_REQUESTED** [review]
Hey @dmgolembiowski - I'm not a maintainer or the owner of this repo but I stumbled across your PR and thought it deserved a look at since you're trying to give back to the community!

Other general notes that don't really fit anywhere:
* Your commit history is messing. Look into using "git interactive rebase" if you're unfamiliar with it and squash your work down into small, atomic commits, that each work by themselves. In my PR (#6), I made one commit for the functionality change, another to add the tests. Something similar would probably make sense here. You'll need to force push your branch to amend things in this PR
* If you use the phrase "Will close" in your original PR message followed by the issue number, GitHub will automagically identify that your PR will close an issue if it's merged, saved the owner or a maintainer having to do that manually

Please note I'm running Windows so I'm not actually able to test your additions, just can help you make your code compile, pass tests, and get this PR tidy ready for reviewing to be merged :)

Also, this is my first time ever code reviewing, so I hope this was helpful!

**Pull request closed without merging**